### DISCUSSION TITLE
When using ListView multiselect how do I manage all selected items?

### DISCUSSION THREAD

**Discussion by Ben2146053** [tui-cs/Terminal.Gui#1518] [Q&A] [answered]
How do I manage the process of getting all selections from a multiselect with ListView?

**Comment by tznind** [comment, marked as answer]
To be clear when you talk about 'getting all selections' you are meaning these right?

![image](https://user-images.githubusercontent.com/31306100/142583089-fffac207-6e6f-41d0-8cda-59ce97ca013a.png)
_Tick indicates marked items_

If you have a list of objects that you provided as the `Source` for the ListView then you can get those that have been multi selected (ticked) by using the `bool IsMarked(int)` method of the source.  It takes the index of the list and returns true/false for whether it is ticked.

So for example if you have a `ListView` called `lv` and a `List` (of data objects) called `myList` you could use the following Linq:

```csharp
myList.Where((e, idx) => lv.Source.IsMarked(idx)) 
```

If you don't have a copy of your original list (i.e. `myList`) you could still get the list of objects with something like (you could probably improve on this!):
```csharp
lv.Source.ToList().Cast<object>().ToList().Where ((e,idx)=>lv.Source.IsMarked(idx))
```
